test/read_write_laspoints.jl

```diff
@@ -0,0 +1,196 @@
+struct TestLasPoint
+    x::Int32
+    y::Int32
+    z::Int32
+    intensity::UInt16
+    flag_byte::UInt8  # return number (3 bits), number of returns (3 bits), scan direction flag, edge of flight line
+    raw_classification::UInt8  # classification (5 bits), classification flags
+    flag_byte_1::UInt8  # return number (4 bits) & number of returns (4 bits)
+    flag_byte_2::UInt8 # classification flags, scanner channel,
+    classification::UInt8
+    user_data::UInt8
+    scan_angle::Int8
+    extended_scan_angle::Int16
+    pt_src_id::UInt16
+    gps_time::Float64
+    red::N0f16
+    green::N0f16
+    blue::N0f16
+    nir::N0f16
+    wave_packet_descriptor_index::UInt8
+    wave_packet_byte_offset::UInt64
+    wave_packet_size_in_bytes::UInt32
+    wave_return_location::Float32
+    wave_x_t::Float32
+    wave_y_t::Float32
+    wave_z_t::Float32
+end
+
+function TestLasPoint()
+    return TestLasPoint(
+        rand(Int32), # x
+        rand(Int32), # y
+        rand(Int32), # z
+        rand(UInt16), # intensity::UInt16
+        rand(UInt8), # flag_byte::UInt8  # return number (3 bits), number of returns (3 bits), scan direction flag, edge of flight line
+        rand(UInt8), # raw_classification::UInt8  # classification (5 bits), classification flags
+        rand(UInt8), # flag_byte_1::UInt8  # return number (4 bits) & number of returns (4 bits)
+        rand(UInt8), # flag_byte_2::UInt8 # classification flags, scanner channel,
+        rand(UInt8), # classification::UInt8
+        rand(UInt8), # user_data::UInt8
+        rand(Int8), # scan_angle::Int16
+        rand(Int16), # scan_angle::Int16
+        rand(UInt16), # pt_src_id::UInt16
+        rand(Float64), # gps_time::Float64
+        rand(N0f16), # red::N0f16
+        rand(N0f16), # green::N0f16
+        rand(N0f16), # blue::N0f16
+        rand(N0f16), # nir::N0f16
+        rand(UInt8), # wave_packet_descriptor_index::UInt8
+        rand(UInt64), # wave_packet_byte_offset::UInt64
+        rand(UInt32), # wave_packet_size_in_bytes::UInt32
+        rand(Float32), # wave_return_location::Float32
+        rand(Float32), # wave_x_t::Float32
+        rand(Float32), # wave_y_t::Float32
+        rand(Float32), # wave_z_t::Float32
+    )
+end
+
+function Base.:(==)(h1::T, h2::T) where T <:Union{LasHeader, LasPoint}
+    # @show typeof(h2)
+    @test typeof(h1) == typeof(h2)
```

Comment:
Better just return false? The outer @test around this equals will then throw at a more appropriate place?
It doesn't matter, the goal is for a failing test in the end...

Reply:
Fixed